### PR TITLE
depr(python,rust!): renaming `approx_unique` as `approx_n_unique`

### PR DESCRIPTION
**Pull request by alexander-beedie**
@ritchie46: as discussed a week or two (or three??) back  ;)

Deprecates the name `approx_unique` in favour of `approx_n_unique`, to better line up with `n_unique` (as, at first glance, `approx_unique` would appear to be an approximate version of `unique`, which is definitely not the case).

### CODE REVIEW

File: crates/polars-ops/src/series/ops/approx_unique.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use polars_core::with_match_physical_integer_polars_type;
 #[cfg(feature = "approx_unique")]
 use crate::series::HyperLogLog;
 
-fn approx_unique_ca<'a, T>(ca: &'a ChunkedArray<T>) -> PolarsResult<Series>
```
Comment (stinodego) [inline crates/polars-ops/src/series/ops/approx_unique.rs:9]: We should probably rename this file as well to `approx_n_unique.rs`

**Review by stinodego — COMMENTED** [review]
Fully agree that this new name is more suitable!

I'm only thinking that maybe `n_unique_approx` would be better due to auto-complete / being more similar to `n_unique`.

File: crates/polars-ops/src/series/ops/approx_unique.rs (hunk 1 above)
Comment (alexander-beedie) [reply crates/polars-ops/src/series/ops/approx_unique.rs:9]: I was wondering about that; one thought - are other any other related algorithms likely to be added (for instance, an _actual_ `approx_unique`?) If so would make sense to keep them in the same place. Though we can always rename now, and simply rename again _if_ that ever happens  ;)

Comment (stinodego) [reply crates/polars-ops/src/series/ops/approx_unique.rs:9]: > Though we can always rename now, and simply rename again if that ever happens ;)

That would generally be my approach. Though if you prefer leaving it as-is, that's also OK with me :)

Comment (alexander-beedie) [reply crates/polars-ops/src/series/ops/approx_unique.rs:9]: As for the name, a quick bit of googling makes me think that most packages that support approximate functions lead with `approx_` (probably to naturally cluster them together in docs/lookups/etc?)

For instance: BigQuery [approx funcs](https://cloud.google.com/bigquery/docs/reference/standard-sql/approximate_aggregate_functions).
Snowflake leads with the algorithm, prefixing their approximate functions with `HLL_`; let's definitely _not_ do that :)

Comment (stinodego) [reply crates/polars-ops/src/series/ops/approx_unique.rs:9]: Let's keep it like this then!

**Review by stinodego — APPROVED** [review]

**Comment by ritchie46** [comment]
Already forgot about that one. :) Thanks!

**Pull request merged**